src/layers/basic.jl

```diff
@@ -418,7 +418,8 @@ function Embedding(in::Integer, out::Integer;
 end
 
 (m::Embedding)(x::Union{OneHotVector, OneHotMatrix}) = m.weight * x # equivalent to m.weight[:,onecold(x)]
```

Comment:
This will hit all the fast paths. You'll need to define the following at L48 of `onehot.jl`.
```julia
const OneHotLikeVector{T, L} = OneHotLike{T, L, 0, 1, T}
const OneHotLikeMatrix{T, L, I} = OneHotLike{T, L, 1, 2, I}
```

```suggestion
(m::Embedding)(x::Union{OneHotLikeVector, OneHotLikeMatrix}) = m.weight * x # equivalent to m.weight[:,onecold(x)]
```

Reply:
Why not use `::OneHotLike` ? Is there reason to constrain it to the two cases?

Reply:
I think the one hot fast path is for vectors and matrices only? So we want to reshape (code a couple lines below) before this.